Added support for LinkBase in Avalonia resources generator

Abdesol opened the pull request:
## What does the pull request do?
adds support for `LinkBase` in avalonia resources generator


## What is the current behavior?
Avalonia asset loader is not resolving for `LinkBase`


## What is the updated/expected behavior with this PR?
Using LinkBase to set resources path works


## Fixed issues
Fixes #16056

Abdesol committed:
added support for linkbase in avalonia resources generator

maxkatz6 committed:
Merge branch 'master' into 16056-fix

kekekeks commented:
Please, add a test (a resource outside of the test lib with LinkBase + some test that loads it from code should be sufficient).

Abdesol committed:
- made the Source class from GenerateAvaloniaResourcesTask internal
- added InternalsVisibleTo of Avalonia.Build.Tasks in Avalonia.Build.Tasks.UnitTest
- added test for LinkBase support

MrJul commented on src/Avalonia.Build.Tasks/GenerateAvaloniaResourcesTask.cs:
```diff
@@ -38,9 +38,12 @@ public Source(ITaskItem avaloniaResourceItem, string root)
                 var path = !string.IsNullOrEmpty(link)
                     ? link
-                    : relativePath;
+                    : !string.IsNullOrEmpty(linkBase)
```
This treats `LinkBase` and `Link` the same, which isn't correct. `Link` is a file path whereas `LinkBase` is the path before the directory globs, i.e. a directory path.

MrJul commented on tests/Avalonia.Build.Tasks.UnitTest/GenerateAvaloniaResourcesTaskTest.cs:
```diff
@@ -0,0 +1,26 @@
+    public void Does_Support_LinkBase_In_Avalonia_Resources_Generator()
+    {
+        var path = "path/to/resources";
+        var expected = "/" + path;
```
The added test doesn't check the right thing: the expected path should be `/path/to/resources/root.xml`.

MrJul reviewed (changes requested):
The current implementation is wrong, see previous comments.

I don't think we should handle `LinkBase` in the task itself.
Let MSBuild expand the recursive globs from `LinkBase` into `Link` instead.

This is what the SDK is doing. We should probably copy this behavior.

Reference: https://github.com/dotnet/sdk/blob/74dafbfb0c0d311595da6fc61d24ca57e6d3f80d/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets#L104-L107

maxkatz6 commented:
Closing due to inactivity
